services/blockchain-connector/shared/sdk/endpoints.js

```diff
@@ -95,6 +106,8 @@ const getBlockByHeight = async (height, includeGenesisAssets = false) => {
 		}
 
 		const block = await invokeEndpoint('chain_getBlockByHeight', { height });
+		await cacheBlocksIfEnabled(block);
```

Comment:
Caching can be done asynchronously. No need to await.
Please do the same everywhere else.
```suggestion
		cacheBlocksIfEnabled(block);
```